src/payment/strategies/adyenv2/adyenv2-payment-strategy.spec.ts

```diff
@@ -242,318 +116,337 @@ describe('AdyenV2PaymentStrategy', () => {
+                }
 
-            jest.spyOn(adyenV2ScriptLoader, 'load').mockReturnValue(Promise.resolve(adyenCheckout));
+                expect(adyenCheckout.create).toHaveBeenCalledTimes(2);
+            });
 
-            jest.spyOn(paymentActionCreator, 'submitPayment')
-                .mockReturnValueOnce(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, identifyShopperError)))
-                .mockReturnValueOnce(submitPaymentAction);
+            it('returns UNKNOWN_ERROR', async () => {
+                jest.spyOn(paymentActionCreator, 'submitPayment')
+                    .mockReturnValueOnce(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, getUnknownError())))
+                    .mockReturnValueOnce(submitPaymentAction);
 
-            try {
                 await strategy.initialize(options);
-                await strategy.execute(getOrderRequestBody());
-            } catch (error) {
-                expect(error).toEqual({
-                    errorCode: 'errorCode',
-                    message: 'errorMessage',
-                });
-            }
-        });
 
-        it('returns 3DS2 IdentifyShopper flow and responses ChallengeShopper flow', async () => {
-            const threeDS2ChallengeComponent: AdyenComponent = {
-                mount: jest.fn(() => {
-                    componentOptions.onComplete({
-                        data: {
-                            threeDS2Token: 'challengeToken',
-                        },
-                    });
+                try {
+                    await strategy.execute(getOrderRequestBody());
+                } catch (error) {
+                    expect(error).toBeInstanceOf(RequestError);
+                }
 
-                    return;
-                }),
-                unmount: jest.fn(),
-            };
+                expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(1);
+                expect(adyenCheckout.create).toHaveBeenCalledTimes(2);
+            });
 
-            jest.spyOn(adyenCheckout, 'create')
-                .mockImplementationOnce(jest.fn((_type, options) => {
-                    componentOptions = options;
+            it('tries to pay with invalid component state', async () => {
+                const adyenInvalidPaymentComponent = {
+                    mount: jest.fn(),
+                    unmount: jest.fn(),
+                };
+                jest.spyOn(adyenCheckout, 'create')
+                    .mockReturnValue(adyenInvalidPaymentComponent);
 
-                    return threeDS2ChallengeComponent;
-                }));
+                await strategy.initialize(options);
 
-            jest.spyOn(adyenV2ScriptLoader, 'load').mockReturnValue(Promise.resolve(adyenCheckout));
+                try {
+                    await strategy.execute(getOrderRequestBody());
+                } catch (error) {
+                    expect(error).toBeInstanceOf(NotInitializedError);
```

Comment:
I don't think this belongs here. I don't think this is being executed.

Reply:
it does, it test

```javascript
                if (!componentState) {
                    throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
                }